internal/docstore/dynamodocstore/dynamo.go

```diff
@@ -100,13 +113,13 @@ func (c *collection) missingKeyField(m map[string]*dyn.AttributeValue) string {
 	return ""
 }
 
-func (c *collection) put(ctx context.Context, doc driver.Document, condition *string) error {
+func (c *collection) put(ctx context.Context, k driver.ActionKind, doc driver.Document, condition *expression.ConditionBuilder) error {
 	av, err := encodeDoc(doc)
 	if err != nil {
 		return err
 	}
 	mf := c.missingKeyField(av.M)
-	if condition != &notExistsCond && mf != "" {
+	if k != driver.Create && mf != "" {
```

Comment:
Instead of passing the Kind, pass a boolean like `createMissingKeyField`. It seems a bit cleaner to me that way.

Reply:
Oh, I thought this was a more direct translation from what was here before. I think from the call site this way it is clear about what value is passed in instead of creating a new boolean.

Reply:
OK.